packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -336,6 +336,52 @@ export const getCartDetails = payload => {
     };
 };
 
+export const retrieveAndMergeCarts = payload =>
+    async function thunk(dispatch) {
+        const { mergeCarts, fetchCartId } = payload;
+        const sourceCartId = await retrieveCartId();
+        try {
+            await dispatch(removeCart());
+            await dispatch(
+                createCart({
+                    fetchCartId
+                })
+            );
+        } catch (error) {
+            dispatch(actions.mergeCarts.receive(error));
+            return thunk(...arguments);
```

Comment:
This will retry the operation right? Could this cause an infinite loop?

Reply:
I followed getDetails as example, I will work on this

Reply:
removed retries